test/translator/common_test.cc

```diff
@@ -150,4 +150,37 @@ TEST(Common, ShouldFailBuildAllocationWhenAllocPageFails) {
   EXPECT_EQ(translator::StatusCode::kFailure, status_code);
 }
 
+TEST(Common, SafePointerCastWrite) {
+  uint32_t expected_val = 0;
+  if (translator::IsLittleEndian())
+    expected_val = 0x13292022;
+  else
+    expected_val = 0x22202913;
+  uint8_t buf[100];
+
+  uint32_t* val = translator::SafePointerCastWrite<uint32_t>(buf);
+  *val = expected_val;
+
+  ASSERT_EQ(0x22, buf[0]);
+  ASSERT_EQ(0x20, buf[1]);
+  ASSERT_EQ(0x29, buf[2]);
+  ASSERT_EQ(0x13, buf[3]);
+}
+
+TEST(Common, SafePointerCastRead) {
+  uint32_t expected_val = 0;
+  if (translator::IsLittleEndian())
+    expected_val = 0x13292022;
+  else
+    expected_val = 0x22202913;
+  uint8_t buf[100];
+  buf[0] = 0x22;
+  buf[1] = 0x20;
+  buf[2] = 0x29;
+  buf[3] = 0x13;
+
+  const uint32_t* val = translator::SafePointerCastRead<uint32_t>(buf);
+  ASSERT_EQ(expected_val, *val);
```

Comment:
This is undefined behavior because it treats a `uint8_t` array as a `uint32_t`, violating strict aliasing. A legal approach would be to instead declare a `uint32_t` to begin with, then use `reinterpret_cast` to convert that into a `uint8_t*` to read from.